src/onedrive.rs

```diff
@@ -269,6 +270,39 @@ impl OneDrive {
             .await
     }
 
+    /// Creates a new [`DriveItem`][drive_item] allowing to set all supported attributes.
+    ///
+    /// # Errors
+    /// Will result in `Err` with HTTP `409 CONFLICT` if [`conflict_behavior`][conflict_behavior]
+    /// is set to [`Fail`][conflict_fail] and the target already exists.
+    ///
+    /// # See also
+    ///
+    /// [Microsoft Docs](https://learn.microsoft.com/en-us/graph/api/resources/driveitem?view=graph-rest-1.0)
+    ///
+    /// [with_opt]: #method.create_folder_with_option
+    /// [drive_item]: ./resource/struct.DriveItem.html
+    /// [conflict_behavior]: ./option/struct.DriveItemPutOption.html#method.conflict_behavior
+    /// [conflict_fail]: ./enum.ConflictBehavior.html#variant.Fail
+    pub async fn create_drive_item<'a, P, D>(
+        &self,
+        parent_item: P,
+        drive_item: D,
+    ) -> Result<DriveItem>
```

Comment:
The conflict behavior is a control option of the method instead of a property of a `DriveItem`, thus it makes more sense to add it here as a parameter rather than inside `DriveItem` struct. 
`#[serde(flatten)]` can be used to merge two structs during serialization. You can follow and/or reuse the impl of [`new_upload_session_with_initial_option`](https://github.com/oxalica/onedrive-api/blob/9c9516818934f4ed79394ee46c0776d5725943f3/src/onedrive.rs#L452).

```suggestion
        drive_item: D,
        conflict_behavior: ConflictBehavior,
    ) -> Result<DriveItem>
```

Also I think we should assert that `drive_item` has `folder` set, or it's a misuse and it will fail anyway.

Reply:
I agree, it makes sense to extract it from the struct.

Regarding the assertion of having `folder` set, I disagree as the `DriveItem` may have another facet. See the comment I've added talking about the facets [here](https://github.com/oxalica/onedrive-api/pull/6#issuecomment-1935800548). We can check that at least one of those four facets is set though.